zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -383,18 +384,53 @@ mod tests {
                 LEFT OUTER JOIN sapling_tree_shards prev_shard
                     ON shard.shard_index = prev_shard.shard_index + 1
                 INNER JOIN scan_queue ON
+                    (scan_queue.block_range_start >= subtree_start_height AND shard.subtree_end_height IS NULL) OR
                     (scan_queue.block_range_start BETWEEN subtree_start_height AND shard.subtree_end_height) OR
                     ((scan_queue.block_range_end - 1) BETWEEN subtree_start_height AND shard.subtree_end_height) OR
```

Comment:
Is it better not to rely on the behaviour of `BETWEEN x and NULL`? As I read the sqlite docs this is always false (as intended), and this appears to be consistent with at least the [X/Open version of the SQL standard](https://pubs.opengroup.org/onlinepubs/9695959099/toc.pdf).

https://www.sqlite.org/lang_expr.html#the_between_operator
> The BETWEEN operator is logically equivalent to a pair of comparisons. "x BETWEEN y AND z" is equivalent to "x>=y AND x<=z" except that with BETWEEN, the x expression is only evaluated once.

https://www.sqlite.org/lang_expr.html#operators_and_parse_affecting_attributes
> All operators generally evaluate to NULL when any operand is NULL, with specific exceptions as stated below. This is in accordance with the SQL92 standard.
>
> When paired with NULL:
> * AND evaluates to 0 (false) when the other operand is false; and
> * OR evaluates to 1 (true) when the other operand is true.

Reply:
We can't rely on the behavior of `OR` with respect to `NULL` here, because we explicitly want to return `TRUE` in the case that `shard.subtree_end_height IS NULL` and none of the other clauses evaluate to `TRUE`.